src/datastore/datastore.rs

```diff
@@ -213,35 +216,40 @@ impl DatastoreWorker {
             loop {
                 let mut request = match self.responder.poll() {
                     Ok(r) => r,
-                    Err(_) => { // All references to responder is gone, quit
-                        info!("DB worker quitting");
+                    Err(e) => { // All references to responder is gone, quit
+                        info!("DB worker quitting due to {:?}", e);
                         self.quit = true;
                         break;
                     }
                 };
                 let response = match request.body() {
                     Commands::CreateBucket(bucket) => {
+                        ds.commit = true;
                         match ds.create_bucket(&transaction, bucket.clone()) {
                             Ok(_) => Ok(Responses::Empty()),
                             Err(e) => Err(e)
                         }
                     },
                     Commands::DeleteBucket(bucketname) => {
+                        ds.commit = true;
                         match ds.delete_bucket(&transaction, bucketname) {
                             Ok(_) => Ok(Responses::Empty()),
                             Err(e) => Err(e)
                         }
                     },
                     Commands::GetBucket(bucketname) => {
+                        ds.commit = true;
                         match ds.get_bucket(bucketname) {
                             Ok(b) => Ok(Responses::Bucket(b)),
                             Err(e) => Err(e)
                         }
                     },
                     Commands::GetBuckets() => {
+                        ds.commit = true;
                         Ok(Responses::BucketMap(ds.get_buckets()))
                     },
                     Commands::InsertEvents(bucketname, events) => {
+                        ds.commit = true;
```

Comment:
Remove

Reply:
I can't until we've fixed the commit-before-get. If I do I have to put `ds.force_commit()` all over the place which is really annoying and hard to debug.